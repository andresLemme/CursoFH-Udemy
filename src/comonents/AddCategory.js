import React, { useState } from "react";
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {
  const [inputValue, setinputValue] = useState('');

  const handleInput = e => {
    setinputValue(e.target.value);
    // console.log(inputValue)
    console.log('handleIInut llamado')
  };
  const handleSubmit = (e) => {
    e.preventDefault()
    
    if(inputValue.trim().length >2){
      setCategories(cats =>[inputValue,...cats]);
      setinputValue('')

    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <h2>AddCategory</h2>
      <input type="text" value={inputValue} onChange={handleInput} />
    </form>
  );
};

AddCategory.PropTypes = {
  setCategories: PropTypes.func.isRequired
}