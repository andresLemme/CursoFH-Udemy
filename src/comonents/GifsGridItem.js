import React from "react";
import PropTypes from 'prop-types';

export const GifsGridItem = ({title, url,id }) => {
  console.log({title,id,url})
  return (
    <div className="card animate__animated animate__bounce">
      <img src={url} alt={title} />
     
      <p>{title}</p>
    </div>
  );

};
GifsGridItem.prototype = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}