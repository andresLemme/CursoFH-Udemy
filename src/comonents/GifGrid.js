import React, { useState, useEffect } from "react";
import { GifsGridItem } from "./GifsGridItem";

export const GifGrid = ({ category }) => {
  const [images, setimages] = useState([]);

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async () => {
    const url =
      "https://api.giphy.com/v1/gifs/search?q=Rick+and+Morty&limit=10&api_key=CMpodpFp15TEuswOPawYe7z29S4MZK5L";
    const resp = await fetch(url);
    const { data } = await resp.json();

    const gifs = data.map(gif => {
      return {
        id: gif.id,
        title: gif.title,
        url: gif.images.downsized_medium.url,
      };
    });

    setimages(gifs);
    // console.log(gifs);
  };

  // getGifs();

  return (
    <div>
      <h3>{category}</h3>

      {images.map(image => (
        <GifsGridItem key={image.id} {...image} />
      ))}
    </div>
  );
};
