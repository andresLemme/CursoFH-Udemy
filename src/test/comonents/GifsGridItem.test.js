import React from "react";
import { shallow } from "enzyme";
import { GifsGridItem } from "../../comonents/GifsGridItem";


describe('Prueba de <GifsGridItem/>', () => {
  test('debe mostrar el comonente correctamente', () => {
     const wrapper = shallow(<GifsGridItem/>)
     expect(wrapper).toMatchSnapshot()
  });
  
});
