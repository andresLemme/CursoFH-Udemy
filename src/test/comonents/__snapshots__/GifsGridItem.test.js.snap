// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de <GifsGridItem/> debe mostrar el comonente correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img />
  <p />
</div>
`;
